status_deprecated.go

```diff
@@ -56,143 +55,67 @@ type StatusDeprecated struct {
 	VerificationErr     error
 }
 
-func FetchStatusDeprecated(f *Ferry, v Verifier) *StatusDeprecated {
-	status := &StatusDeprecated{}
+func FetchStatus(f *Ferry, v Verifier) *Status {
```

Comment:
If this is moved to collocate with the ControlServer, you can then attach this method directly on the ControlServer and then remove the need for the arguments.